### ISSUE TITLE
test kitchen not running

### ISSUE THREAD

**Issue by axos88** [schubergphilis/chef-acme#53]
on ubuntu the server does not start up correcly, due to a wrong passowrd (and etchosts file too)
on centos it doesn't find the box.

```
      
       
       These processes exited early (check above for their output):
        'GORACE="halt_on_error=1" exec ./bin/boulder-sa --config test/boulder-config.json' with pid 25461 exited 1
       started exec listenbuddy -listen :5673 -speak boulder-rabbitmq:5672 with pid 24823
       export GO15VENDOREXPERIMENT=1
       GOBIN=/opt/go/src/github.com/letsencrypt/boulder/bin go install  ./...
       started GORACE="halt_on_error=1" exec ./bin/boulder-wfe --config test/boulder-config.json with pid 25457
       started GORACE="halt_on_error=1" exec ./bin/boulder-ra --config test/boulder-config.json with pid 25458
       started GORACE="halt_on_error=1" exec ./bin/boulder-sa --config test/boulder-config.json with pid 25461
       started GORACE="halt_on_error=1" exec ./bin/boulder-ca --config test/boulder-config.json with pid 25464
       started GORACE="halt_on_error=1" exec ./bin/boulder-va --config test/boulder-config.json with pid 25470
       I212423 boulder-wfe Versions: boulder-wfe=(Unspecified Unspecified) Golang=(go1.5) BuildHost=(Unspecified)
       started GORACE="halt_on_error=1" exec ./bin/boulder-publisher --config test/boulder-config.json with pid 25473
       I212423 boulder-ra Versions: boulder-ra=(Unspecified Unspecified) Golang=(go1.5) BuildHost=(Unspecified)
       started GORACE="halt_on_error=1" exec ./bin/ocsp-updater --config test/boulder-config.json with pid 25476
       started GORACE="halt_on_error=1" exec ./bin/ocsp-responder --config test/boulder-config.json with pid 25482
       I212424 boulder-sa Versions: boulder-sa=(Unspecified Unspecified) Golang=(go1.5) BuildHost=(Unspecified)
       I212424 boulder-wfe WFE using key policy: goodkey.KeyPolicy{AllowRSA:true, AllowECDSANISTP256:true, AllowECDSANISTP384:true, AllowECDSANISTP521:false}
       I212424 boulder-wfe Server running, listening on 0.0.0.0:4000...
       
       I212424 boulder-ra loading hostname policy, sha256: 3ddc932620e3f59623af52971fcc3cf38c9759f64575fd018a7097875427ea5e
       started GORACE="halt_on_error=1" exec ./bin/ct-test-srv --config test/boulder-config.json with pid 25484
       started GORACE="halt_on_error=1" exec ./bin/dns-test-srv --config test/boulder-config.json with pid 25487
       I212424 boulder-ca Versions: boulder-ca=(Unspecified Unspecified) Golang=(go1.5) BuildHost=(Unspecified)
       I212424 boulder-ca loading hostname policy, sha256: 3ddc932620e3f59623af52971fcc3cf38c9759f64575fd018a7097875427ea5e
       started GORACE="halt_on_error=1" exec ./bin/mail-test-srv --config test/boulder-config.json with pid 25489
       E212424 boulder-sa [AUDIT] Couldn't connect to SA database: Error 1045: Access denied for user 'sa'@'localhost' (using password: NO)
       Couldn't connect to SA database: Error 1045: Access denied for user 'sa'@'localhost' (using password: NO)
       started GORACE="halt_on_error=1" exec ./bin/ocsp-responder --config test/issuer-ocsp-responder.json with pid 25493
       I212424 boulder-va Versions: boulder-va=(Unspecified Unspecified) Golang=(go1.5) BuildHost=(Unspecified)
       I212424 boulder-publisher Versions: boulder-publisher=(Unspecified Unspecified) Golang=(go1.5) BuildHost=(Unspecified)
       started GORACE="halt_on_error=1" exec ./bin/caa-checker --config cmd/caa-checker/test-config.yml with pid 25499
       
       
       These processes exited early (check above for their output):
        'GORACE="halt_on_error=1" exec ./bin/boulder-sa --config test/boulder-config.json' with pid 25461 exited 1

```

```
akos@akos-ui ~/projects/freelancer/chef/chef-acme $ kitchen converge ssl-centos-7
Expected array default value for '--driver'; got "kitchen-vagrant" (string)
-----> Starting Kitchen (v1.14.0)
-----> Creating <ssl-centos-7>...
       Bringing machine 'default' up with 'virtualbox' provider...
       ==> default: Box 'opscode-centos-7.1' could not be found. Attempting to find and install...
           default: Box Provider: virtualbox
           default: Box Version: >= 0
       ==> default: Box file was not detected as metadata. Adding it directly...
       ==> default: Adding box 'opscode-centos-7.1' (v0) for provider: virtualbox
           default: Downloading: opscode-centos-7.1
An error occurred while downloading the remote file. The error
       message, if any, is reproduced below. Please fix this error and try
       again.
       
       Couldn't open file /home/akos/projects/freelancer/chef/chef-acme/.kitchen/kitchen-vagrant/kitchen-chef-acme-ssl-centos-7/opscode-centos-7.1
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [Expected process to exit with [0], but received '1'
---- Begin output of vagrant up --no-provision --provider virtualbox ----
STDOUT: Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'opscode-centos-7.1' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Box file was not detected as metadata. Adding it directly...
==> default: Adding box 'opscode-centos-7.1' (v0) for provider: virtualbox
    default: Downloading: opscode-centos-7.1

STDERR: An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

Couldn't open file /home/akos/projects/freelancer/chef/chef-acme/.kitchen/kitchen-vagrant/kitchen-chef-acme-ssl-centos-7/opscode-centos-7.1
---- End output of vagrant up --no-provision --provider virtualbox ----
Ran vagrant up --no-provision --provider virtualbox returned 1] on ssl-centos-7
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

**Comment by SpikesDivZero** [comment]
tl;dr: I don't know how to fix the CentOS issue, but I'm going to at least try and fix the Ubuntu kitchen issue.

For CentOS, I was able to partially fix this with with patch:

```patch
index c6c192a..b651232 100644
--- a/.kitchen.yml
+++ b/.kitchen.yml
@@ -26,7 +26,7 @@ driver:
 platforms:
 - name: centos-7
   driver:
-    box: opscode-centos-7.1 # virtualbox
+    box: bento/centos-7.1   # virtualbox
     image: centos-7-0-x64   # digitalocean
 - name: ubuntu-14.04
   run_list:
```

However, it now errors with an error that I'm not sure how best to fix [acme-53-centos-error.txt](https://github.com/schubergphilis/chef-acme/files/629893/acme-53-centos-error.txt)

This looks to be due to the resolved erlang cookbook being quite outdated. To be sure, I completely removed my ~/.berkshelf/ folder and retried, and it still errored. [acme-53-Berksfile.lock.txt](https://github.com/schubergphilis/chef-acme/files/629892/acme-53-Berksfile.lock.txt) contains the resolved dependency details.

For Ubuntu, I'm getting a similar issue with nothing propagating into /etc/hosts. It looks like this traces back to a few commits in the boulder project, all relating to letsencrypt/boulder#1639, with the two most notable being:

letsencrypt/boulder@d98eb634d134863717b91e348a09182eb7020c6a
```patch
diff --git a/README.md b/README.md
index 049df95..491d8bc 100644
--- a/README.md
+++ b/README.md
@@ -57,6 +57,10 @@ or
 
 (On OS X, using port, you will have to add `CGO_CFLAGS="-I/opt/local/include" CGO_LDFLAGS="-L/opt/local/lib"` to your environment or `go` invocations.)
 
+Edit /etc/hosts to add this line:
+
+    127.0.0.1 boulder-rabbitmq boulder-mysql
+
 Resolve Go-dependencies, set up a database and RabbitMQ:
 
     ./test/setup.sh
```

and

letsencrypt/boulder@dfc1f9465c284fd1f7492ba1b7c3734b51b750da
```patch
diff --git a/test/setup.sh b/test/setup.sh
index 1581a1a..21d1f74 100755
--- a/test/setup.sh
+++ b/test/setup.sh
@@ -24 +24 @@ go get \
-go run cmd/rabbitmq-setup/main.go -server amqp://localhost &
+go run cmd/rabbitmq-setup/main.go -server amqp://boulder-rabbitmq &
```

**Comment by SpikesDivZero** [comment]
Ah, nevermind my last comment. I missed that you had uploaded PR #54.

**Comment by thoutenbos** [comment]
Should be fixed now for CentOS!
